app/controllers/users_controller.rb

```diff
@@ -135,12 +135,14 @@ def getting_started_completed
     redirect_to stream_path
   end
 
-  def export
-    if export = Diaspora::Exporter.new(current_user).execute
-      send_data export, filename: "#{current_user.username}_diaspora_data.json", type: :json
-    else
-      head :not_acceptable
-    end
+  def export_profile
+    current_user.queue_export
+    flash[:notice] = I18n.t('users.edit.export_in_progress')
+    redirect_to edit_user_path
+  end
+
+  def download_profile
+    send_data decompressed_export, type: :json, filename: current_user.export.filename
```

Comment:
Or maybe make it a tarball, even most Windows users should have a tool for that, I think 7-zip is quite common and can handle those.